src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -132,20 +132,27 @@ internal partial class AnalyzerManager
         }
 
         /// <summary>
-        /// Get all the analyzer actions to execute for the given analyzer against a given compilation.
+        /// Get tuple with:
+        /// (a) All the analyzer actions to execute for the given analyzer against a given compilation and
+        /// (b) A flag indicating if these actions can be executed concurrently.
         /// The returned actions include the actions registered during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/> method as well as
         /// the actions registered during <see cref="CompilationStartAnalyzerAction"/> for the given compilation.
         /// </summary>
-        public async Task<AnalyzerActions> GetAnalyzerActionsAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
+        public async Task<Tuple<AnalyzerActions, bool>> GetAnalyzerActionsAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
In this case, is there a problem with just splitting it into two calls?  Looks like you just need the `sessionScope` to get the bool.  Is calling `GetSessionAnalysisScopeAsync` twice a problem?